flightsql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -0,0 +1,1916 @@
+         */
+        @VisibleForTesting
+        static final TableDefinition DEFINITION = TableDefinition.of(
+                ColumnDefinition.ofString(CATALOG_NAME),
+                ColumnDefinition.ofString(DB_SCHEMA_NAME),
+                ColumnDefinition.ofString(TABLE_NAME), // out-of-spec
+                ColumnDefinition.ofString(COLUMN_NAME), // out-of-spec
+                ColumnDefinition.ofString(KEY_NAME),
+                ColumnDefinition.ofInt(KEY_SEQUENCE) // out-of-spec
+        );
+
+        private static final Map<String, Object> ATTRIBUTES = Map.of();
+        private static final Table TABLE = TableTools.newTable(DEFINITION, ATTRIBUTES);
+    }
+
+    private boolean hasTable(String catalog, String dbSchema, String table) {
+        if (catalog != null && !catalog.isEmpty()) {
+            return false;
+        }
+        if (dbSchema != null && !dbSchema.isEmpty()) {
+            return false;
+        }
+        final Object obj;
+        {
+            final QueryScope scope = ExecutionContext.getContext().getQueryScope();
+            try {
+                obj = scope.readParamValue(table);
+            } catch (QueryScope.MissingVariableException e) {
+                return false;
+            }
+        }
+        if (!(obj instanceof Table)) {
+            return false;
+        }
+        return authorization.transform((Table) obj) != null;
+    }
+
+    private final CommandHandler commandGetPrimaryKeysHandler = new CommandStaticTable<>(CommandGetPrimaryKeys.class,
+            CommandGetPrimaryKeysConstants.TABLE, FlightSqlTicketHelper::ticketFor) {
+        @Override
+        void checkForGetInfo(CommandGetPrimaryKeys command) {
+            if (CommandGetPrimaryKeys.getDefaultInstance().equals(command)) {
+                // TODO: Plumb through io.deephaven.server.arrow.FlightServiceGrpcImpl.getSchema
+                // We need to pretend that CommandGetPrimaryKeys.getDefaultInstance() is a valid command until we can
+                // plumb getSchema through to the resolvers.
+                return;
+            }
+            if (!hasTable(
+                    command.hasCatalog() ? command.getCatalog() : null,
+                    command.hasDbSchema() ? command.getDbSchema() : null,
+                    command.getTable())) {
+                throw tableNotFound();
+            }
+        }
+
+        // No need to check at resolve time since there is no actual state involved. If Deephaven exposes the notion
+        // of keys, this will need to behave more like QueryBase where there is a handle-based ticket and some sort
+        // state maintained. It is also incorrect to perform the same checkForFlightInfo at resolve time because the
+        // state of the server may have changed between getInfo and doGet/doExchange, and getInfo should still be valid
+        // for client.
+        // @Override
+        // void checkForResolve(CommandGetPrimaryKeys command) {
+        // }
```

Comment:
Is there a todo of some sort that we can add? Or maybe this just needs a little cleaning now that you're further along. Or maybe you just don't want the method commented anymore? Perhaps the abstract parent should not have a default impl so each subclass is forced to consider it?

Reply:
Yep, will add reference